src/Module/Conversation/Channel.php

```diff
@@ -0,0 +1,398 @@
+			self::$item_id = $item['parent-uri-id'] ?? 0;
+		} else {
+			self::$item_id = 0;
+		}
+
+		self::$min_id = $_GET['min_id']       ?? null;
+		self::$max_id = $_GET['max_id']       ?? null;
+		self::$max_id = $_GET['last_created'] ?? self::$max_id;
+	}
+
+	/**
+	 * Computes the displayed items.
+	 *
+	 * Community pages have a restriction on how many successive posts by the same author can show on any given page,
+	 * so we may have to retrieve more content beyond the first query
+	 *
+	 * @return array
+	 * @throws \Exception
+	 */
+	protected static function getItems()
+	{
+		if (self::$content == self::WHATSHOT) {
+			if (!is_null(self::$accountType)) {
+				$condition = ["(`comments` >= ? OR `activities` >= ?) AND `contact-type` = ?", self::getMedianComments(4), self::getMedianActivities(4), self::$accountType];
+			} else {
+				$condition = ["(`comments` >= ? OR `activities` >= ?) AND `contact-type` != ?", self::getMedianComments(4), self::getMedianActivities(4), Contact::TYPE_COMMUNITY];
+			}
+		} elseif (self::$content == self::FORYOU) {
+			$cid = Contact::getPublicIdByUserId(DI::userSession()->getLocalUserId());
+
+			$condition = ["(`owner-id` IN (SELECT `relation-cid` FROM `contact-relation` WHERE `cid` = ? AND `thread-score` > ?) OR
+				((`comments` >= ? OR `activities` >= ?) AND `owner-id` IN (SELECT `pid` FROM `account-user-view` WHERE `uid` = ? AND `rel` IN (?, ?))) OR
+				( `owner-id` IN (SELECT `pid` FROM `account-user-view` WHERE `uid` = ? AND `rel` IN (?, ?) AND `notify_new_posts`)))",
+				$cid, self::getMedianThreadScore($cid, 4), self::getMedianComments(4), self::getMedianActivities(4), DI::userSession()->getLocalUserId(), Contact::FRIEND, Contact::SHARING,
+				DI::userSession()->getLocalUserId(), Contact::FRIEND, Contact::SHARING];
+		} elseif (self::$content == self::FOLLOWERS) {
+			$condition = ["`owner-id` IN (SELECT `pid` FROM `account-user-view` WHERE `uid` = ? AND `rel` = ?)", DI::userSession()->getLocalUserId(), Contact::FOLLOWER];
+		} elseif (self::$content == self::IMAGE) {
+			$condition = ["`media-type` & ?", 1];
+		} elseif (self::$content == self::VIDEO) {
+			$condition = ["`media-type` & ?", 2];
+		} elseif (self::$content == self::AUDIO) {
+			$condition = ["`media-type` & ?", 4];
+		}
+
+		$condition[0] .= " AND NOT EXISTS(SELECT `cid` FROM `user-contact` WHERE `uid` = ? AND `cid` = `post-engagement`.`owner-id` AND (`ignored` OR `blocked` OR `collapsed`))";
+		$condition[] = DI::userSession()->getLocalUserId();
+
+		if ((self::$content != self::WHATSHOT) && !is_null(self::$accountType)) {
+			$condition[0] .= " AND `contact-type` = ?";
+			$condition[] = self::$accountType;
+		}
+
+		$params = ['order' => ['created' => true], 'limit' => self::$itemsPerPage];
+
+		if (!empty(self::$item_id)) {
+			$condition[0] .= " AND `uri-id` = ?";
+			$condition[] = self::$item_id;
+		} else {
+			if (!empty($_REQUEST['no_sharer'])) {
```

Comment:
Please use `$request` instead of the `$_REQUEST` superglobal. You will have to pass it to `getItems` as a parameter.

Reply:
Done